.github/workflows/build-web-extension.yml

```diff
@@ -0,0 +1,45 @@
+name: Build Web Extension
+
+on:
+    workflow_dispatch:
+
+    push:
+        branches:
```

Comment:
I'm not sure if you wanted to "restrict" automatic building to just the `master' (and `v4`, for now) branches, since it's a slow thing, and we have the option to build manually for any branch (via `workflow_dispatch` above).

Reply:
I think for a trigger semver tags make the most sense (e.g. `v4.0.0`). This gives us more control over when the build should be triggered plus it gives us a release to attach the artifacts to.

Reply:
Thanks for sharing your plan! A couple of things to note:

1. The artifacts are stored for up to 90 days, so they're not a good candidate/alternative for linking release/downloads. There's a download-artifact action which we can use to copy and store artifacts somewhere "permanently".
2. The best advantage of CI/CD is to really have these things be built to be tested easily on interim steps (the "continuous" part of the word), not just releases. I'd even say the release part is the least important here, and why I'd like to make it a separate GitHub action that only runs manually (unless we're releasing a new version "every day" or continuously -- in that case, it makes more sense to have it triggered automatically too).

So I think this will be good enough for us to test (when needed) the things we're working on, and then the release "button" can be pushed whenever there's a new release to be made.

Let me know if that makes sense or not, and if there's anything unclear about it.